File: pg-client2/src/pg/oid.clj
(ns pg.oid
  (:import
   com.github.igrishaev.enums.OID))

(def ^OID default OID/DEFAULT)
(def ^OID bool OID/BOOL)
(def ^OID _bool OID/_BOOL)
(def ^OID bytea OID/BYTEA)
(def ^OID _bytea OID/_BYTEA)
(def ^OID char OID/CHAR)
(def ^OID _char OID/_CHAR)
(def ^OID name OID/NAME)
(def ^OID _name OID/_NAME)
(def ^OID int8 OID/INT8)
(def ^OID _int8 OID/_INT8)
(def ^OID int2 OID/INT2)
(def ^OID _int2 OID/_INT2)
(def ^OID int2vector OID/INT2VECTOR)
(def ^OID _int2vector OID/_INT2VECTOR)
(def ^OID int4 OID/INT4)
(def ^OID _int4 OID/_INT4)
(def ^OID regproc OID/REGPROC)
(def ^OID _regproc OID/_REGPROC)
(def ^OID text OID/TEXT)
(def ^OID _text OID/_TEXT)
(def ^OID oid OID/OID)
(def ^OID _oid OID/_OID)
(def ^OID tid OID/TID)
(def ^OID _tid OID/_TID)
(def ^OID xid OID/XID)
(def ^OID _xid OID/_XID)
(def ^OID cid OID/CID)
(def ^OID _cid OID/_CID)
(def ^OID oidvector OID/OIDVECTOR)
(def ^OID _oidvector OID/_OIDVECTOR)
(def ^OID pg_type OID/PG_TYPE)
(def ^OID _pg_type OID/_PG_TYPE)
(def ^OID pg_attribute OID/PG_ATTRIBUTE)
(def ^OID _pg_attribute OID/_PG_ATTRIBUTE)
(def ^OID pg_proc OID/PG_PROC)
(def ^OID _pg_proc OID/_PG_PROC)
(def ^OID pg_class OID/PG_CLASS)
(def ^OID _pg_class OID/_PG_CLASS)
(def ^OID json OID/JSON)
(def ^OID _json OID/_JSON)
(def ^OID xml OID/XML)
(def ^OID _xml OID/_XML)
(def ^OID pg_node_tree OID/PG_NODE_TREE)
(def ^OID pg_ndistinct OID/PG_NDISTINCT)
(def ^OID pg_dependencies OID/PG_DEPENDENCIES)
(def ^OID pg_mcv_list OID/PG_MCV_LIST)
(def ^OID pg_ddl_command OID/PG_DDL_COMMAND)
(def ^OID xid8 OID/XID8)
(def ^OID _xid8 OID/_XID8)
(def ^OID point OID/POINT)
(def ^OID _point OID/_POINT)
(def ^OID lseg OID/LSEG)
(def ^OID _lseg OID/_LSEG)
(def ^OID path OID/PATH)
(def ^OID _path OID/_PATH)
(def ^OID box OID/BOX)
(def ^OID _box OID/_BOX)
(def ^OID polygon OID/POLYGON)
(def ^OID _polygon OID/_POLYGON)
(def ^OID line OID/LINE)
(def ^OID _line OID/_LINE)
(def ^OID float4 OID/FLOAT4)
(def ^OID _float4 OID/_FLOAT4)
(def ^OID float8 OID/FLOAT8)
(def ^OID _float8 OID/_FLOAT8)
(def ^OID unknown OID/UNKNOWN)
(def ^OID circle OID/CIRCLE)
(def ^OID _circle OID/_CIRCLE)
(def ^OID money OID/MONEY)
(def ^OID _money OID/_MONEY)
(def ^OID macaddr OID/MACADDR)
(def ^OID _macaddr OID/_MACADDR)
(def ^OID inet OID/INET)
(def ^OID _inet OID/_INET)
(def ^OID cidr OID/CIDR)
(def ^OID _cidr OID/_CIDR)
(def ^OID macaddr8 OID/MACADDR8)
(def ^OID _macaddr8 OID/_MACADDR8)
(def ^OID aclitem OID/ACLITEM)
(def ^OID _aclitem OID/_ACLITEM)
(def ^OID bpchar OID/BPCHAR)
(def ^OID _bpchar OID/_BPCHAR)
(def ^OID varchar OID/VARCHAR)
(def ^OID _varchar OID/_VARCHAR)
(def ^OID date OID/DATE)
(def ^OID _date OID/_DATE)
(def ^OID time OID/TIME)
(def ^OID _time OID/_TIME)
(def ^OID timestamp OID/TIMESTAMP)
(def ^OID _timestamp OID/_TIMESTAMP)
(def ^OID timestamptz OID/TIMESTAMPTZ)
(def ^OID _timestamptz OID/_TIMESTAMPTZ)
(def ^OID interval OID/INTERVAL)
(def ^OID _interval OID/_INTERVAL)
(def ^OID timetz OID/TIMETZ)
(def ^OID _timetz OID/_TIMETZ)
(def ^OID bit OID/BIT)
(def ^OID _bit OID/_BIT)
(def ^OID varbit OID/VARBIT)
(def ^OID _varbit OID/_VARBIT)
(def ^OID numeric OID/NUMERIC)
(def ^OID _numeric OID/_NUMERIC)
(def ^OID refcursor OID/REFCURSOR)
(def ^OID _refcursor OID/_REFCURSOR)
(def ^OID regprocedure OID/REGPROCEDURE)
(def ^OID _regprocedure OID/_REGPROCEDURE)
(def ^OID regoper OID/REGOPER)
(def ^OID _regoper OID/_REGOPER)
(def ^OID regoperator OID/REGOPERATOR)
(def ^OID _regoperator OID/_REGOPERATOR)
(def ^OID regclass OID/REGCLASS)
(def ^OID _regclass OID/_REGCLASS)
(def ^OID regcollation OID/REGCOLLATION)
(def ^OID _regcollation OID/_REGCOLLATION)
(def ^OID regtype OID/REGTYPE)
(def ^OID _regtype OID/_REGTYPE)
(def ^OID regrole OID/REGROLE)
(def ^OID _regrole OID/_REGROLE)
(def ^OID regnamespace OID/REGNAMESPACE)
(def ^OID _regnamespace OID/_REGNAMESPACE)
(def ^OID uuid OID/UUID)
(def ^OID _uuid OID/_UUID)
(def ^OID pg_lsn OID/PG_LSN)
(def ^OID _pg_lsn OID/_PG_LSN)
(def ^OID tsvector OID/TSVECTOR)
(def ^OID _tsvector OID/_TSVECTOR)
(def ^OID gtsvector OID/GTSVECTOR)
(def ^OID _gtsvector OID/_GTSVECTOR)
(def ^OID tsquery OID/TSQUERY)
(def ^OID _tsquery OID/_TSQUERY)
(def ^OID regconfig OID/REGCONFIG)
(def ^OID _regconfig OID/_REGCONFIG)
(def ^OID regdictionary OID/REGDICTIONARY)
(def ^OID _regdictionary OID/_REGDICTIONARY)
(def ^OID jsonb OID/JSONB)
(def ^OID _jsonb OID/_JSONB)
(def ^OID jsonpath OID/JSONPATH)
(def ^OID _jsonpath OID/_JSONPATH)
(def ^OID txid_snapshot OID/TXID_SNAPSHOT)
(def ^OID _txid_snapshot OID/_TXID_SNAPSHOT)
(def ^OID pg_snapshot OID/PG_SNAPSHOT)
(def ^OID _pg_snapshot OID/_PG_SNAPSHOT)
(def ^OID int4range OID/INT4RANGE)
(def ^OID _int4range OID/_INT4RANGE)
(def ^OID numrange OID/NUMRANGE)
(def ^OID _numrange OID/_NUMRANGE)
(def ^OID tsrange OID/TSRANGE)
(def ^OID _tsrange OID/_TSRANGE)
(def ^OID tstzrange OID/TSTZRANGE)
(def ^OID _tstzrange OID/_TSTZRANGE)
(def ^OID daterange OID/DATERANGE)
(def ^OID _daterange OID/_DATERANGE)
(def ^OID int8range OID/INT8RANGE)
(def ^OID _int8range OID/_INT8RANGE)
(def ^OID int4multirange OID/INT4MULTIRANGE)
(def ^OID _int4multirange OID/_INT4MULTIRANGE)
(def ^OID nummultirange OID/NUMMULTIRANGE)
(def ^OID _nummultirange OID/_NUMMULTIRANGE)
(def ^OID tsmultirange OID/TSMULTIRANGE)
(def ^OID _tsmultirange OID/_TSMULTIRANGE)
(def ^OID tstzmultirange OID/TSTZMULTIRANGE)
(def ^OID _tstzmultirange OID/_TSTZMULTIRANGE)
(def ^OID datemultirange OID/DATEMULTIRANGE)
(def ^OID _datemultirange OID/_DATEMULTIRANGE)
(def ^OID int8multirange OID/INT8MULTIRANGE)
(def ^OID _int8multirange OID/_INT8MULTIRANGE)
(def ^OID record OID/RECORD)
(def ^OID _record OID/_RECORD)
(def ^OID cstring OID/CSTRING)
(def ^OID _cstring OID/_CSTRING)
(def ^OID any OID/ANY)
(def ^OID anyarray OID/ANYARRAY)
(def ^OID void OID/VOID)
(def ^OID trigger OID/TRIGGER)
(def ^OID event_trigger OID/EVENT_TRIGGER)
(def ^OID language_handler OID/LANGUAGE_HANDLER)
(def ^OID internal OID/INTERNAL)
(def ^OID anyelement OID/ANYELEMENT)
(def ^OID anynonarray OID/ANYNONARRAY)
(def ^OID anyenum OID/ANYENUM)
(def ^OID fdw_handler OID/FDW_HANDLER)
(def ^OID index_am_handler OID/INDEX_AM_HANDLER)
(def ^OID tsm_handler OID/TSM_HANDLER)
(def ^OID table_am_handler OID/TABLE_AM_HANDLER)
(def ^OID anyrange OID/ANYRANGE)
(def ^OID anycompatible OID/ANYCOMPATIBLE)
(def ^OID anycompatiblearray OID/ANYCOMPATIBLEARRAY)
(def ^OID anycompatiblenonarray OID/ANYCOMPATIBLENONARRAY)
(def ^OID anycompatiblerange OID/ANYCOMPATIBLERANGE)
(def ^OID anymultirange OID/ANYMULTIRANGE)
(def ^OID anycompatiblemultirange OID/ANYCOMPATIBLEMULTIRANGE)
(def ^OID pg_brin_bloom_summary OID/PG_BRIN_BLOOM_SUMMARY)
(def ^OID pg_brin_minmax_multi_summary OID/PG_BRIN_MINMAX_MULTI_SUMMARY)

File: pg-client2/src/com/github/igrishaev/enums/OID.java
package com.github.igrishaev.enums;

import com.github.igrishaev.PGError;

import java.util.HashMap;
import java.util.Map;

public enum OID {
    DEFAULT                      (   0),
    BOOL                         (  16),
    _BOOL                        (1000),
    BYTEA                        (  17),
    _BYTEA                       (1001),
    CHAR                         (  18),
    _CHAR                        (1002),
    NAME                         (  19),
    _NAME                        (1003),
    INT8                         (  20),
    _INT8                        (1016),
    INT2                         (  21),
    _INT2                        (1005),
    INT2VECTOR                   (  22),
    _INT2VECTOR                  (1006),
    INT4                         (  23),
    _INT4                        (1007),
    REGPROC                      (  24),
    _REGPROC                     (1008),
    TEXT                         (  25),
    _TEXT                        (1009),
    OID                          (  26),
    _OID                         (1028),
    TID                          (  27),
    _TID                         (1010),
    XID                          (  28),
    _XID                         (1011),
    CID                          (  29),
    _CID                         (1012),
    OIDVECTOR                    (  30),
    _OIDVECTOR                   (1013),
    PG_TYPE                      (  71),
    _PG_TYPE                     ( 210),
    PG_ATTRIBUTE                 (  75),
    _PG_ATTRIBUTE                ( 270),
    PG_PROC                      (  81),
    _PG_PROC                     ( 272),
    PG_CLASS                     (  83),
    _PG_CLASS                    ( 273),
    JSON                         ( 114),
    _JSON                        ( 199),
    XML                          ( 142),
    _XML                         ( 143),
    PG_NODE_TREE                 ( 194),
    PG_NDISTINCT                 (3361),
    PG_DEPENDENCIES              (3402),
    PG_MCV_LIST                  (5017),
    PG_DDL_COMMAND               (  32),
    XID8                         (5069),
    _XID8                        ( 271),
    POINT                        ( 600),
    _POINT                       (1017),
    LSEG                         ( 601),
    _LSEG                        (1018),
    PATH                         ( 602),
    _PATH                        (1019),
    BOX                          ( 603),
    _BOX                         (1020),
    POLYGON                      ( 604),
    _POLYGON                     (1027),
    LINE                         ( 628),
    _LINE                        ( 629),
    FLOAT4                       ( 700),
    _FLOAT4                      (1021),
    FLOAT8                       ( 701),
    _FLOAT8                      (1022),
    UNKNOWN                      ( 705),
    CIRCLE                       ( 718),
    _CIRCLE                      ( 719),
    MONEY                        ( 790),
    _MONEY                       ( 791),
    MACADDR                      ( 829),
    _MACADDR                     (1040),
    INET                         ( 869),
    _INET                        (1041),
    CIDR                         ( 650),
    _CIDR                        ( 651),
    MACADDR8                     ( 774),
    _MACADDR8                    ( 775),
    ACLITEM                      (1033),
    _ACLITEM                     (1034),
    BPCHAR                       (1042),
    _BPCHAR                      (1014),
    VARCHAR                      (1043),
    _VARCHAR                     (1015),
    DATE                         (1082),
    _DATE                        (1182),
    TIME                         (1083),
    _TIME                        (1183),
    TIMESTAMP                    (1114),
    _TIMESTAMP                   (1115),
    TIMESTAMPTZ                  (1184),
    _TIMESTAMPTZ                 (1185),
    INTERVAL                     (1186),
    _INTERVAL                    (1187),
    TIMETZ                       (1266),
    _TIMETZ                      (1270),
    BIT                          (1560),
    _BIT                         (1561),
    VARBIT                       (1562),
    _VARBIT                      (1563),
    NUMERIC                      (1700),
    _NUMERIC                     (1231),
    REFCURSOR                    (1790),
    _REFCURSOR                   (2201),
    REGPROCEDURE                 (2202),
    _REGPROCEDURE                (2207),
    REGOPER                      (2203),
    _REGOPER                     (2208),
    REGOPERATOR                  (2204),
    _REGOPERATOR                 (2209),
    REGCLASS                     (2205),
    _REGCLASS                    (2210),
    REGCOLLATION                 (4191),
    _REGCOLLATION                (4192),
    REGTYPE                      (2206),
    _REGTYPE                     (2211),
    REGROLE                      (4096),
    _REGROLE                     (4097),
    REGNAMESPACE                 (4089),
    _REGNAMESPACE                (4090),
    UUID                         (2950),
    _UUID                        (2951),
    PG_LSN                       (3220),
    _PG_LSN                      (3221),
    TSVECTOR                     (3614),
    _TSVECTOR                    (3643),
    GTSVECTOR                    (3642),
    _GTSVECTOR                   (3644),
    TSQUERY                      (3615),
    _TSQUERY                     (3645),
    REGCONFIG                    (3734),
    _REGCONFIG                   (3735),
    REGDICTIONARY                (3769),
    _REGDICTIONARY               (3770),
    JSONB                        (3802),
    _JSONB                       (3807),
    JSONPATH                     (4072),
    _JSONPATH                    (4073),
    TXID_SNAPSHOT                (2970),
    _TXID_SNAPSHOT               (2949),
    PG_SNAPSHOT                  (5038),
    _PG_SNAPSHOT                 (5039),
    INT4RANGE                    (3904),
    _INT4RANGE                   (3905),
    NUMRANGE                     (3906),
    _NUMRANGE                    (3907),
    TSRANGE                      (3908),
    _TSRANGE                     (3909),
    TSTZRANGE                    (3910),
    _TSTZRANGE                   (3911),
    DATERANGE                    (3912),
    _DATERANGE                   (3913),
    INT8RANGE                    (3926),
    _INT8RANGE                   (3927),
    INT4MULTIRANGE               (4451),
    _INT4MULTIRANGE              (6150),
    NUMMULTIRANGE                (4532),
    _NUMMULTIRANGE               (6151),
    TSMULTIRANGE                 (4533),
    _TSMULTIRANGE                (6152),
    TSTZMULTIRANGE               (4534),
    _TSTZMULTIRANGE              (6153),
    DATEMULTIRANGE               (4535),
    _DATEMULTIRANGE              (6155),
    INT8MULTIRANGE               (4536),
    _INT8MULTIRANGE              (6157),
    RECORD                       (2249),
    _RECORD                      (2287),
    CSTRING                      (2275),
    _CSTRING                     (1263),
    ANY                          (2276),
    ANYARRAY                     (2277),
    VOID                         (2278),
    TRIGGER                      (2279),
    EVENT_TRIGGER                (3838),
    LANGUAGE_HANDLER             (2280),
    INTERNAL                     (2281),
    ANYELEMENT                   (2283),
    ANYNONARRAY                  (2776),
    ANYENUM                      (3500),
    FDW_HANDLER                  (3115),
    INDEX_AM_HANDLER             ( 325),
    TSM_HANDLER                  (3310),
    TABLE_AM_HANDLER             ( 269),
    ANYRANGE                     (3831),
    ANYCOMPATIBLE                (5077),
    ANYCOMPATIBLEARRAY           (5078),
    ANYCOMPATIBLENONARRAY        (5079),
    ANYCOMPATIBLERANGE           (5080),
    ANYMULTIRANGE                (4537),
    ANYCOMPATIBLEMULTIRANGE      (4538),
    PG_BRIN_BLOOM_SUMMARY        (4600),
    PG_BRIN_MINMAX_MULTI_SUMMARY (4601);

    private final int code;

    private final static Map<Integer, OID> intToItem;

    static {
        intToItem = new HashMap<>();
        for (OID oid: values()) {
            intToItem.put(oid.code, oid);
        }
    }

    OID(final int code) {
        this.code = code;
    }

    public static OID ofInt (final int code) {
        final OID result = intToItem.get(code);
        if (result == null) {
            throw new PGError("unknown OID, code: %s", code);
        }
        else {
            return result;
        }
    }

    public int toInt() {
        return code;
    }

}
